tests/unit/storage/data/external/ExternalClients.js

```diff
@@ -154,6 +156,62 @@ describe('external backend clients', () => {
                     });
             });
         });
+
+        if (backend.config.type !== 'azure') {
+            it(`${backend.name} should set tags and then delete it`, done => {
+                const key = 'externalBackendTestKey';
+                const bucket = {
+                    getName: sinon.fake.returns('externalBackendTestBucket'),
+                };
+                const objectMd = {
+                    tags: {
+                        Key1: 'value_1',
+                        Key2: 'value_2',
+                    },
+                    location: [
+                        {
+                            dataStoreVersionId: 'latestversion',
+                        },
+                    ],
+                };
+                async.series([
+                    next => testClient.objectPutTagging(key.key, bucket, objectMd, log, next),
+                    next => testClient.objectDeleteTagging(key.Key, bucket, objectMd, log, next),
+                ], done);
+            });
+
+            it(`${backend.name} should fail to set tag on missing key`, done => {
+                const key = 'externalBackendMissingKey';
+                const bucket = {
+                    getName: sinon.fake.returns('externalBackendTestBucket'),
+                };
+                const objectMD = {
+                    tags: {
+                        Key1: 'value_1',
+                    },
+                    location: [
+                        {
+                            dataStoreVersionId: 'latestversion',
+                        },
+                    ],
+                };
+                async.series(
+                    [
+                        next => testClient.objectPutTagging(key, bucket, objectMD, log, (err) => {
+                            assert(err);
```

Comment:
You can technically remove this assert as it is included in the next one.

Reply:
```suggestion
```

Reply:
I am not too sure, here the first `assert` checks that `err` is not `undefined` or `null`. then the second `assert` check that `err.ServiceUnavailable` is not `undefined` or `null`.
If i remove the first `assert` and `err` is `null` then trying to reach a property of a `null` object would fail.

Reply:
If you try and access a property that is undefined you get an exception which is a test failure so that is fine.
Basically: `assert.strictEqual(a.b.c.d.e.myProperty, 5)` is effectively testing:

```js
assert.notStrictEqual(a, undefined);
assert.notStrictEqual(a.b, undefined);
assert.notStrictEqual(a.b.c, undefined);
assert.notStrictEqual(a.b.c.d, undefined);
assert.notStrictEqual(a.b.c.d.e, undefined);
assert.strictEqual(a.b.c.d.e.myProperty, 5);
```

Because for myProperty to equal 5 all the parent attributes must also exist. So one assertion here is fine.

Reply:
Alright, I will update the test and remove the asserts but I maintain that raising an exception due to trying to access the property of an `undefined` object is different that raising an assert because the object is undefined. but this is very very minor differences.